airbyte-integrations/connectors/source-amplitude/components.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from dataclasses import dataclass
+from typing import Any, Dict, List, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+
+from airbyte_cdk.models import FailureType
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from airbyte_cdk.sources.declarative.migrations.state_migration import StateMigration
+from airbyte_cdk.sources.declarative.schema import JsonFileSchemaLoader
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.declarative.types import Config, Record
+from airbyte_cdk.sources.types import Config, StreamSlice, StreamState
+from airbyte_cdk.utils import AirbyteTracedException
+
+
+logger = logging.getLogger("airbyte")
+
+
+class AverageSessionLengthRecordExtractor(RecordExtractor):
+    """
+    Create records from complex response structure
+    Issue: https://github.com/airbytehq/airbyte/issues/23145
```

Comment:
Since the docs link to the same issue, I'd love to have longer, more descriptive docstrings that explain what the custom component does, and why do we have to use it. 

I know this might not be related to the migration itself as you're just moving them over.